api/routes.py

```diff
@@ -95,40 +95,53 @@ def decorated(*args, **kwargs):
         return f(*args, **kwargs)
     return decorated
 
-# The allows_patron_web decorator will add Cross-Origin Resource Sharing
+# The allows_cors decorator will add Cross-Origin Resource Sharing
 # (CORS) headers to routes that will be used by the patron web interface.
 # This is necessary for a JS app on a different domain to make requests.
 #
 # This is mostly taken from the cross_origin decorator in flask_cors, but we
 # can't use that decorator because we aren't able to look up the patron web
 # client url configuration setting at the time we create the decorator.
-def allows_patron_web(f):
+def allows_cors(allowed_domain_type):
     # Override Flask's default behavior and intercept the OPTIONS method for
     # every request so CORS headers can be added.
-    f.required_methods = getattr(f, 'required_methods', set())
-    f.required_methods.add("OPTIONS")
-    f.provide_automatic_options = False
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            func.required_methods = getattr(func, 'required_methods', set())
+            func.required_methods.add("OPTIONS")
+            func.provide_automatic_options = False
 
-    def wrapped_function(*args, **kwargs):
-        if request.method == "OPTIONS":
-            resp = app.make_default_options_response()
-        else:
-            resp = make_response(f(*args, **kwargs))
+            if request.method == "OPTIONS":
+                resp = app.make_default_options_response()
+            else:
+                resp = make_response(func(*args, **kwargs))
+            
+            patron_web_domains = app.manager.patron_web_domains
+            admin_web_domains = app.manager.admin_web_domains
+            
+            if "patron" in allowed_domain_type:
+                web_domains = " ,".join(admin_web_domains.union(patron_web_domains))
+            else:
+                web_domains = admin_web_domains
```

Comment:
Since you treat `allowed_domain_type` as a collection, it looks as though calling this code will set the patron web domains but not the admin web domains:

```
@allows_cors(allowed_domain_type=set({"patron"}))
```

But actually, both sets of domains will be allowed.

Since your calls to @allows_cors are already explicit about which domains are allowed, I think it's best to make this method explicit as well. Add the patron_web_domains to a set if 'patron' is one of the allowed types, and then add the admin_web_domains to that set if 'admin' is one of the allowed types.

Apart from this minor issue I think this looks good and is ready to merge.

Reply:
Ah, yes, this should be made more clear. I pushed up a change that I think is better. Let me know if this is what you were thinking.

Reply:
This looks great but your commit also introduced some Dockerfile changes which should go into another branch (from Nick?). Take those out and this is good to merge.

Reply:
Good catch. I have unintentionally committed those files about 5 times on this PR 😵‍💫 . Reverted!